projectns/cs_listgroupchans.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023 Ryan Schmidt
+ * Rights to this code are as documented in doc/LICENSE.
+ *
+ * Services awareness of group registrations
+ * ChanServ command for contacts to list channels within their namespaces
+ */
+
+#include "fn-compat.h"
+#include "atheme.h"
+#include "projectns.h"
+
+static void cmd_listgroupchans(sourceinfo_t *si, int parc, char *parv[]);
+
+command_t cs_listgroupchans = { "LISTGROUPCHANS", N_("List channels belonging to your projects."), AC_AUTHENTICATED, 2, cmd_listgroupchans, { .path = "freenode/cs_listgroupchans" } };
+
+static void cmd_listgroupchans(sourceinfo_t *si, int parc, char *parv[])
+{
+	char *filter = parv[0];
+
+	mowgli_node_t *n;
+	mowgli_list_t *plist = projectsvs->myuser_get_projects(si->smu);
+	unsigned int num_projects = 0;
+
+	MOWGLI_ITER_FOREACH(n, plist->head)
+		num_projects++;
+
+	if (!num_projects)
```

Comment:
This can be simplified to

```C
    if (! MOWGLI_LIST_LENGTH(plist))
```

Reply:
Done, albeit without the space between the ! operator and the macro. If this is required for atheme coding conventions, I'll add that space in.